src/main/scala-2/chisel3/simulator/HasSimulator.scala

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.simulator
+
+import java.nio.file.Files
+
+/** Type class for providing a simulator. */
+trait HasSimulator {
+
+  /** Return a simulator.  This require providing a type class implementation of
+    * [[HasTestingDirectory]].
+    */
+  def getSimulator(implicit testingDirectory: HasTestingDirectory): Simulator[_]
+
+}
+
+/** Type class implementations of [[HasSimulator]]. */
+object HasSimulator {
+
+  /** A [[HasSimulator]] implementation for a Verilator simulator. */
+  val verilator: HasSimulator = new HasSimulator {
```

Comment:
Similarly to the below, is there any real reason to make these `val` vs. `def`?

Reply:
No reason. Changed to a def in a312ee898.